### PR TITLE
stdio/file: unify group file list to common single queue

### PR DESCRIPTION
**Pull request by anchao**
## Summary

stdio/file: unify group file list to common single queue

unify group file list to common single queue

Signed-off-by: chao an <anchao@lixiang.com>

## Impact

N/A

## Testing

ci-check

### CODE REVIEW

File: include/nuttx/fs/fs.h [hunk 1]
```diff
@@ -538,7 +538,7 @@ struct filelist
 #ifdef CONFIG_FILE_STREAM
 struct file_struct
 {
-  FAR struct file_struct *fs_next;      /* Pointer to next file stream */
+  sq_entry_t              fs_entry;     /* Entry of file stream */
```
Comment (xiaoxiang781216) [inline include/nuttx/fs/fs.h:541]: should we include queue.h in this file and remove from other source file?

File: libs/libc/stdio/lib_libflushall.c [hunk 2]
```diff
@@ -28,6 +28,8 @@
 #include <fcntl.h>
 #include <errno.h>
 
+#include <nuttx/nuttx.h>
```
Comment (xiaoxiang781216) [inline libs/libc/stdio/lib_libflushall.c:31]: why need nuttx/nuttx.h

File: sched/tls/task_initinfo.c [hunk 3]
```diff
@@ -25,6 +25,7 @@
 #include <errno.h>
 #include <fcntl.h>
 
+#include <nuttx/nuttx.h>
```
Comment (xiaoxiang781216) [inline sched/tls/task_initinfo.c:28]: ditto

File: sched/tls/task_uninitinfo.c [hunk 4]
```diff
@@ -22,9 +22,11 @@
  * Included Files
  ****************************************************************************/
 
+#include <nuttx/nuttx.h>
 #include <nuttx/kmalloc.h>
 #include <nuttx/mutex.h>
 #include <nuttx/list.h>
+#include <nuttx/queue.h>
```
Comment (xiaoxiang781216) [inline sched/tls/task_uninitinfo.c:29]: remove

**Review by acassis — APPROVED** [review]

File: sched/tls/task_uninitinfo.c (hunk 4 above)
Comment (anchao) [reply sched/tls/task_uninitinfo.c]: Done

File: sched/tls/task_initinfo.c (hunk 3 above)
Comment (anchao) [reply sched/tls/task_initinfo.c]: Done

File: libs/libc/stdio/lib_libflushall.c (hunk 2 above)
Comment (anchao) [reply libs/libc/stdio/lib_libflushall.c:31]: Done

File: include/nuttx/fs/fs.h (hunk 1 above)
Comment (anchao) [reply include/nuttx/fs/fs.h:542]: Done

File: libs/libc/stdio/lib_libflushall.c (hunk 2 above)
Comment (xiaoxiang781216) [reply libs/libc/stdio/lib_libflushall.c:31]: but no change

Comment (anchao) [reply libs/libc/stdio/lib_libflushall.c:31]: This file requires the implementation of container_of()

Comment (xiaoxiang781216) [reply libs/libc/stdio/lib_libflushall.c:31]: but nobody call container_of in this file?

Comment (anchao) [reply libs/libc/stdio/lib_libflushall.c:31]: see line 71 and 124

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**